app/base/file_handler.py

```diff
@@ -17,40 +21,57 @@ def create_images_folder(username):
     return property_img_folder
 
 
-def property_image_handler(username, image_files, save_to_folder):
+def save_images_to_temp_folder(image_files):
     """
-    Handles the docs uploaded from the web form. The docs are down sized using
-    the Pillow image library and saved to the file system. The image filenames are checked
-    using the werkzeug utilities, then the filename is saved to the list of filenames in the
-    dictionary object.
+    Generates a temporary folder for storing image files of property.
     """
-    images_list = [save_to_folder]
+    rand_dir_name = secrets.token_hex(16)
+    temp_dir = rand_dir_name
+    os.mkdir(f"{current_app.root_path}/base/static/property_images/{temp_dir}")
+
     for image_file in image_files:
+        filename_hex = secrets.token_hex(10)
         checked_filename = secure_filename(image_file.filename)
-        random_hex = secrets.token_hex(15)
-        _, file_extension = os.path.splitext(checked_filename)
-        clean_filename = f"{username}{random_hex}{file_extension}"
-        images_list.append(clean_filename)
+        _, file_ext = os.path.splitext(checked_filename)
+        checked_filename = f"{filename_hex}{file_ext}"
+        image_file.save(f"{current_app.root_path}/base/static/property_images/{temp_dir}/{checked_filename}")
 
-        image = Image.open(image_file)
-        # reduce image size down to 15%
-        image.thumbnail((3000, 3000))
+    return f"property_images/{temp_dir}"
 
-        file_path = os.path.join(
-            f"{current_app.root_path}/base/static/{save_to_folder}", clean_filename
-        )
-        image.save(file_path)
-    return images_list
 
+def property_image_handler(temp_img_folder=None):
+    """
+    Handles the images uploaded from the web form. The images are down sized using
+    the Pillow image library and saved to the file system. The image filenames are checked
+    using the werkzeug utilities, then the filename is appended to the list of filenames.
+    """
+    # Delete images before before the property is updated with new images
+    rand_dir_name = secrets.token_hex(16)
+    save_to_folder = f"{current_app.root_path}/base/static/property_images/{rand_dir_name}"
+    os.mkdir(save_to_folder)
+    images_list = [f"property_images/{rand_dir_name}"]
+    tmp_dir = f"{current_app.root_path}/base/static/{temp_img_folder}"
 
-def save_profile_picture(username, form_picture):
+    for image_filename in os.listdir(tmp_dir):
+        images_list.append(image_filename)
+        image_path = f"{tmp_dir}/{image_filename}"
+        image_file = Image.open(image_path)
+        image_file.thumbnail((3000, 3000))
+        path = f"{save_to_folder}/{image_filename}"
+        image_file.save(path)
```

Comment:
This is what a celery worker is meant for
lets take this task to a worker
imagine we had 2000 users and each made an upload of 300 images which would mean 
600,000 images to be processed

you  wouldnt want a regular function to do that

a worker would be great to handle this as a cronjob